impeller/entity/contents/filters/blend_filter_contents.cc

```diff
@@ -864,6 +853,29 @@ std::optional<Entity> BlendFilterContents::CreateFramebufferAdvancedBlend(
 
   std::shared_ptr<CommandBuffer> cmd_buffer =
       renderer.GetContext()->CreateCommandBuffer();
+
+  // Generate a 1x1 texture to implement foreground color blending.
+  if (foreground_color.has_value()) {
```

Comment:
This is better because we don't need to create an extra cmd buffer or host texture for the upload.